core/src/main/scala/cats/instances/function.scala

```diff
@@ -81,7 +81,10 @@ sealed private[instances] trait Function0Instances extends Function0Instances0 {
         () => (fa(), fb())
 
       override def ap[A, B](f: () => A => B)(fa: () => A): () => B =
-        () => f().apply(fa())
```

Comment:
why was that `.apply` here in the first place? does `() => f()(fa())` work on dotty, maybe?

Reply:
I think it starts to look like a soup of parens, so I didn't write it that way.